heatmap/src/heatmap.rs

```diff
@@ -196,40 +209,54 @@ impl Heatmap {
         self.summary.buckets()
     }
 
+    /// Returns the number of valid and active `Histogram` slices in the `Heatmap`
+    pub fn active_slices(&self) -> usize {
+        if Instant::now() - self.first_tick > self.span {
+            self.slices()
+        } else {
+            self.idx.load(Ordering::Relaxed) + 1
+        }
+    }
+
+    /// Returns the `DateTime` representation of when the `Heatmap` was created
+    pub fn created_at(&self) -> DateTime {
+        self.create_at
+    }
+
     /// Increment a time-value pair by a specified count
-    pub fn increment(&self, time: Instant, value: u64, count: u32) {
+    pub fn increment(&self, time: Instant, value: u64, count: u32) -> Result<(), Error> {
         self.tick(time);
 
-        let current = self.current.load(Ordering::Relaxed);
+        let curr_tick = self.curr_tick.load(Ordering::Relaxed);
+        let idx = self.idx.load(Ordering::Relaxed);
 
         // fast path when the time falls into the current time slice
-        if time >= self.slices[current].start.load(Ordering::Relaxed) {
+        if time >= curr_tick {
             let _ = self.summary.increment(value, count);
```

Comment:
Since we're returning a `Result` now - it would be better to change this to:

```
self.summary.increment(value, count)?;
self.histograms[idx].increment(value, count)?;
return Ok(());
```

This means that we'll get automatic conversion of the histogram errors into heatmap errors if the value is too large. Currently, this would not return an error if the value is outside of the storable range

Reply:
fixed.